File: TestCPP/Hello.swift
//
    /*******************************************************************************
            
            File name:     File.swift
            Author:        FeiYehua
            
            Description:
            
            History:
                    2024/3/17: File created.
            
    ********************************************************************************/
    

import Foundation
import CxxStdlib
class SayHello{
    /*func sayHelloSwift(to:String) -> String{
        //let cString = to
        let result=String(cString: sayHello(to.cString(using: .utf8)))
        //Hello();
        return result;
    }*/
    func sayHelloCPP(to:String) -> String{
        let cString = std.string(to)
        let result = sayHelloInCPPString(cString)
        return String(result)
    }
}

File: TestCPP/hello.cpp
/*** 
 * @Author       : FeiYehua
 * @Date         : 2024-03-09 21:27:19
 * @LastEditTime : 2024-03-17 02:00:58
 * @LastEditors  : FeiYehua
 * @Description  : 
 * @FilePath     : hello.cpp
 * @     © 2021 FeiYehua
 */
#include "hello.hpp"
using namespace std;
char* sayHello(const char *inputString)
{
    char* str = new char[200];
    strcat(str,"Hello,");
    strcat(str, inputString);
    return str;
}
string sayHelloInCPPString(string inputString)
{
    return "Hello, " + inputString + "From Cpp String";
}
void Hello()
{
    //freopen("w","OutPut.txt",stdout);
    //cout<<"hello,world!"<<endl;
    return ;
}
